src/fsharp/CompilerOptions.fs

```diff
@@ -287,80 +287,80 @@ let ParseCompilerOptions (collectOtherArgument: string -> unit, blocks: Compiler
 
         let rec attempt l =
           match l with
-          | (CompilerOption(s, _, OptionHelp f, d, _) :: _) when optToken = s  && argString = "" ->
+          | CompilerOption(s, _, OptionHelp f, d, _) :: _ when optToken = s  && argString = "" ->
               reportDeprecatedOption d
               f blocks; t
-          | (CompilerOption(s, _, OptionUnit f, d, _) :: _) when optToken = s  && argString = "" ->
+          | CompilerOption(s, _, OptionUnit f, d, _) :: _ when optToken = s  && argString = "" ->
               reportDeprecatedOption d
               f (); t
-          | (CompilerOption(s, _, OptionSwitch f, d, _) :: _) when getSwitchOpt optToken = s && argString = "" ->
+          | CompilerOption(s, _, OptionSwitch f, d, _) :: _ when getSwitchOpt optToken = s && argString = "" ->
               reportDeprecatedOption d
               f (getSwitch opt); t
-          | (CompilerOption(s, _, OptionSet f, d, _) :: _) when optToken = s && argString = "" ->
+          | CompilerOption(s, _, OptionSet f, d, _) :: _ when optToken = s && argString = "" ->
               reportDeprecatedOption d
               f := true; t
-          | (CompilerOption(s, _, OptionClear f, d, _) :: _) when optToken = s && argString = "" ->
+          | CompilerOption(s, _, OptionClear f, d, _) :: _ when optToken = s && argString = "" ->
               reportDeprecatedOption d
               f := false; t
-          | (CompilerOption(s, _, OptionString f, d, _) as compilerOption :: _) when optToken = s ->
+          | CompilerOption(s, _, OptionString f, d, _) as compilerOption :: _ when optToken = s ->
               reportDeprecatedOption d
               let oa = getOptionArg compilerOption argString
               if oa <> "" then
                   f (getOptionArg compilerOption oa)
               t
-          | (CompilerOption(s, _, OptionInt f, d, _) as compilerOption :: _) when optToken = s ->
+          | CompilerOption(s, _, OptionInt f, d, _) as compilerOption :: _ when optToken = s ->
```

Comment:
Precedence of `as` and `::` are subtle here.  But there's no mistake in removal

Reply:
@dsyme Thanks! I think about tweaking the analyzer, so it's not that strict in cases like this one where precedence might be hard to grasp.

Reply:
If it's possible to change the tooling to always require parens around `::` patterns (one for a sequence of them) that would be good, though doesn't have to be part of this PR.

Reply:
I'll try to play with it to find the line where we should not suggest removing parens, yes.